CI/publish_deps.ps1

```diff
@@ -5,22 +5,35 @@ $ErrorView = "NormalView"
 $src_dir=$pwd
 $COMPILER=$Env:COMPILER
 
-if ($COMPILER -eq "Visual Studio 16 2019") {
-	cd 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\14.29.30133\x64\Microsoft.VC142.CRT'
-	cp .\msvcp140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
-	cp .\vcruntime140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
-}else {
-	cd 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Redist\MSVC\14.29.30133\x64\Microsoft.VC142.CRT'
-	cp .\msvcp140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
-	cp .\vcruntime140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
-}
-
 cd $src_dir
 mkdir dependencies
 cd dependencies
 wget http://swdownloads.analog.com/cse/build/libiio-win-deps-libusb1.0.24.zip -OutFile "libiio-win-deps.zip"
 7z x -y "libiio-win-deps.zip"
 
-cp .\libs\64\libxml2.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
-cp .\libs\64\libserialport-0.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
-cp .\libs\64\libusb-1.0.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+if ($COMPILER -eq "MinGW Makefiles") {
+	cp $src_dir\dependencies\libs\64\libserialport-0.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp $src_dir\dependencies\libs\64\libusb-1.0.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp C:\ghcup\ghc\9.2.8\mingw\bin\libgcc_s_seh-1.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp C:\ghcup\ghc\9.4.5\mingw\bin\libiconv-2.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp C:\ghcup\ghc\9.4.5\mingw\bin\zlib1.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp C:\ghcup\ghc\9.4.5\mingw\bin\liblzma-5.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp C:\ghcup\ghc\9.4.5\mingw\bin\libwinpthread-1.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp C:\ghcup\ghc\9.4.5\mingw\bin\libxml2-2.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp C:\ghcup\ghc\9.2.8\mingw\bin\libstdc++-6.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+}
+else {
+	cp $src_dir\dependencies\libs\64\libxml2.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp $src_dir\dependencies\libs\64\libserialport-0.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	cp $src_dir\dependencies\libs\64\libusb-1.0.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+
+	if ($COMPILER -eq "Visual Studio 16 2019") {
+		cd 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\14.29.30133\x64\Microsoft.VC142.CRT'
+		cp .\msvcp140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+		cp .\vcruntime140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	}else {
+		cd 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Redist\MSVC\14.29.30133\x64\Microsoft.VC142.CRT'
+		cp .\msvcp140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+		cp .\vcruntime140.dll $env:BUILD_ARTIFACTSTAGINGDIRECTORY
+	}
```

Comment:
You differenciate Visual Studio 2019 vs. 2022?

If that's a valid change, it definitely does not belong in a commit named "add MinGW builds".

Reply:
Differentiation between Visual Studio 2019 and 2022 was there even before.
Now I just added an `if` statement for MinGW Makefiles, and because of the required files on MinGW vs. VS a restructuring was done for libserialport-0.dll, libusb-1.0.dll, and libxml2.dll.

Reply:
Ok, I didn't notice that you moved the code from lines 8-17 there. I thought you added this.